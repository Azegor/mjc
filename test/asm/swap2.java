class Foobar {
    public int a;

    public void inc() {
        a = a + 1;
    }

    public static void main(String[] args){
      Foobar f1 = new Foobar();
      Foobar f2 = new Foobar();

      f1.a = 0;
      f2.a = 999;

      int i = 0;
      while(i < 1) {
        Foobar tmp = f1;
        f1 = f2;
        f2 = tmp;

        f2.inc();

        i = i + 1;
        System.out.println(1337);
      }

      System.out.println(f1.a);
      System.out.println(f2.a);
    }
}
