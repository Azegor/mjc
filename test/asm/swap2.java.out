1337
999
1
